packages/amplify-graphql-relational-transformer/src/resolvers.ts

```diff
@@ -61,8 +61,9 @@ export const updateTableForReferencesConnection = (
   // -- ideally further up the chain
   const partitionKeyType = attributeTypeFromType(referenceNode.type, ctx);
 
-  // TODO: Add support for sortKey in GSI
-  const sortKey = undefined;
+  // The sortKeys are any referencesNodes following the first element.
+  const sortKeyReferenceNodes = referenceNodes.slice(1);
```

Comment:
what if this doesn't exist? Would add some unit tests to cover case of none, one and more than one sort key fields.

Reply:
Referring to the later, but taking a deeper look, the `addGlobalSecondaryIndex` does handle [undefined sortKey](https://github.com/aws-amplify/amplify-category-api/blob/162e83dd9ff34c6c72b4cffe3ca7059fb235b303/packages/amplify-graphql-relational-transformer/src/resolvers.ts#L175).